### DISCUSSION TITLE
[antd: Notification] You are calling notice in render which will break in React 18 concurrent mode. Please trigger in effect instead.

### DISCUSSION THREAD

**Discussion by saifali40** [ant-design/ant-design#45792] [Q&A]
Warning: [antd: Notification] You are calling notice in render which will break in React 18 concurrent mode. Please trigger in effect instead.

i'm getting this error while using notification / Message.

**Comment by saifali40** [comment]
```import React from 'react';
import { App } from 'antd';

const MyPage: React.FC = () => {
    const { message, notification, modal } = App.useApp();
    message.success('Good!');
    notification.info({ message: 'Good' });
    modal.warning({ title: 'Good' });
    // ....
    // other message, notification, modal static function
    return <div>Hello word</div>;
};

const MyApp: React.FC = () => (
    <App>
        <MyPage />
    </App>
);

export default MyApp;

```

the above code itself breaking.

**Comment by emanuelvsz** [comment]
I'm having the sabe problem, i want to know a way to fix it too. :(

**Comment by afc163** [comment]
Please trigger in effect instead.

**Reply by nerdess** [reply]
I am triggering it in useEffect but still get the error msg...

This is the chain of things:

React Context > triggers hook to grab data > triggers hook that displays an antd message while data is loading.

Inside this custom hook all is wrapped inside an affect:

```
import { ReactNode, useEffect, useRef } from 'react';
import { message } from 'antd';
import { MessageType } from 'antd/lib/message/interface';

const useStatusMessage = ({
	isLoading,
	isError,
	messages,
}: {
	isLoading: boolean;
	isError: boolean;
	messages: {
		isLoading?: string | ReactNode;
		isError?: string | ReactNode;
	};
}): void => {
	
	const info = useRef<MessageType>();
	const [messageApi] = message.useMessage();

	useEffect(() => {
		const { isLoading: loadingMessage, isError: errorMessage } = messages;

		if (isError && !!errorMessage) {
			info.current = messageApi.error({
				content: errorMessage,
			});
			return;
		}

		if (isLoading && !!loadingMessage) {
			info.current = messageApi.loading({
				content: loadingMessage,
				duration: 0,
			});
			return;
		}

		if (!!info.current && !isLoading) info.current();

	}, [messageApi, isError, isLoading, messages]);
};

export default useStatusMessage;

```

What could be the fix?

**Reply by lucabubi** [reply]
instead of messageApi. try using directly message.

**Reply by nerdess** [reply]
@lucabubi But why can I not use messageApi? Also, messageApi is much more versatile (I can add a key and so on...)

**Reply by deepak1-1** [reply]
@nerdess Any Update over it 
how to use new version of messageApi instead of using directly message ?

**Reply by nethzzz** [reply]
https://github.com/ant-design/ant-design/discussions/45792?sort=old#discussioncomment-9158583

**Comment by nethzzz** [comment]
Use {cotextHolder} on the top your return
[here](https://github.com/ant-design/ant-design/issues/39592#issuecomment-1354112707)

**Comment by irfandyj** [comment]
Any update on the issue? This still happens.

**Comment by irfandyj** [comment]
The documentation is out of date, you can directly use `notification.error`, or any directly without calling `notification.useNotification` first.

**Comment by lkzwc** [comment]
hook版本用不了嘛？】
